dephell_archive/_stream.py

```diff
@@ -35,14 +52,24 @@ def is_file(self) -> bool:
         # zip
         return info.filename[-1] != '/'
 
+    def _is_implicit_dir(self) -> bool:
+        # Only zip have implicit dirs
+        if not hasattr(self.descriptor, 'namelist'):
+            return False
+        if self._dir_list is None:
+            self._dir_list = _dir_list(self.descriptor.namelist())
+        path = self.member_path.as_posix()
+        return path in self._dir_list
+
     def is_dir(self) -> bool:
         info = self._get_info()
         if info is None:
-            return False
+            return self._is_implicit_dir()
```

Comment:
I would be also happier, maybe even happier, to have the virtual dirs only implemented in ArchivePath, and not in ArchiveStream .

Reply:
But for that to be sensible, an ArchivePath for a virtual dir shouldnt use the ArchiveStream at all, and it would probably be a good idea to rename ArchiveStream members to not appear to be implementing Path class concepts.